internal/consensus/state.go

```diff
@@ -1442,7 +1429,9 @@ func (cs *State) enterPrecommit(height int64, round int32) {
 		return
 	}
 
-	// If +2/3 prevoted for proposal block, stage and precommit it
+	// If greater than 2/3 of the voting power on the network prevoted for
+	// the proposed block, update our locked block to this block and issue a
+	// precommit vote for it.
```

Comment:
What about unlocking logic at lines 1460-1462? In case there is a non-nil polka for a block we don't have, we currently unlock. I think we should also remove these lines together with L1469. In this scenario it is sufficient to precommit nil and update `cs.ProposalBlock` and `cs.ProposalBlockParts` so we start fetching the block for which there is a polka.

Reply:
Fair enough, I was planning to remove this in a follow up PR, but I'm happy to just remove it all now.